src/http_connection.cpp

```diff
@@ -530,13 +530,28 @@ void http_connection::on_resolve(error_code const& e
 
 	aux::random_shuffle(m_endpoints.begin(), m_endpoints.end());
 
-	// sort the endpoints so that the ones with the same IP version as our
-	// bound listen socket are first. So that when contacting a tracker,
-	// we'll talk to it from the same IP that we're listening on
-	if (m_bind_addr != address_v4::any())
-		std::partition(m_endpoints.begin(), m_endpoints.end()
+	// if we have been told to bind to a particular address
+	// only connect to addresses of the same family
+	if (m_bind_addr)
+	{
+		auto new_end = std::partition(m_endpoints.begin(), m_endpoints.end()
```

Comment:
We generate separate announces for each one of the _client's_ listen sockets. This is concerning the set of addresses the tracker(s) are listening on so the logic still applies. If we generated separate announces for each tracker address there would be M*N announces for each listen socket and tracker address which I don't think is what we want.